src/FakeItEasy/ExceptionThrowerConfigurationExtensions.cs

```diff
@@ -130,6 +134,11 @@ public static IAfterCallSpecifiedConfiguration Throws<T1>(this IExceptionThrower
                 });
         }
 
+#if DISABLED
+
+        // This method already has a generic parameter, and it can't be inferred; if we add another, we make it unusable,
+        // especially since the new one is something unintelligible like IAnyCallConfigurationWithReturnTypeSpecificied<Foo>...
+

```

Comment:
> > I've always seen these extension methods as coding convenience, not necessity.

> Well, they are a convenience, but without them, the API becomes quite inconvenient...

What I meant was that any API endpoint that we provide via extension method we could also have provided using traditional, non-extension methods. Implementing via extension method is just something that reduces a little of the FakeItEasy code.

> Yes, but we don't expose the classes directly, only the interfaces... so the method wouldn't be available (unless we make it part of the interface, but that would make it harder to implement)

Absolutely.

> But in other cases I still have to add a type param

Ah, so I just got lucky by using this particular case as a test. Too bad.